--- js.c	2025-11-05 16:55:43.307576100 +0000
+++ fixed-js.c	2025-11-15 22:33:57.878319000 +0000
@@ -145,7 +145,7 @@
     if (current.type == TOKEN_LET) {
         next_token();
         if (current.type != TOKEN_ID) { printf("Expected identifier\n"); exit(1); }
-        char name[64]; strcpy(name, current.text);
+        char name[64]; strncpy(name, current.text, sizeof(name)-1); name[sizeof(name)-1] = '\0';
         next_token();
         if (current.type != TOKEN_ASSIGN) { printf("Expected '='\n"); exit(1); }
         next_token();
